### ISSUE TITLE
zpool attach/labelclear deny function (even with -f) + zpool split fake error message

### ISSUE THREAD

**Issue by GregorKopka** [zfsonlinux/zfs#9586]
### System information

Type | Version/Name
 --- | --- 
Distribution Name	|  Funtoo
Distribution Version	|  rolling, current
Linux Kernel	| 4.19.67_p2-r1-debian-sources-lts
Architecture	| x86_64 Zen2
ZFS Version	| 0.8.1-r0-gentoo
SPL Version	| 0.8.1-r0-gentoo

### Describe the problem you're observing
-f switches without function:
zpool attach of partition already containing zfs member not possible
zpool labelclear of partition containing zfs member not possible

### Describe how to reproduce the problem
**Idea is to copy an active USB boot pool (sda) onto a drive (sdb) in server, _again_.**
**Thus the remnants of the last iteration are still sitting on sdb1**

**Let's make it a mirror of the booted-from USB drive to later split off the local pool drive:**
$ zpool attach bootstick /dev/sda1 /dev/sdb1
invalid vdev specification
use '-f' to override the following errors:
/dev/sdb1 is part of active pool 'bootstick'

**Fine then. Yes I'm sure, off you go:**
$ zpool attach -f bootstick /dev/sda1 /dev/sdb1
invalid vdev specification
the following errors must be manually repaired:
/dev/sdb1 is part of active pool 'bootstick'

**OK, _then_ let's wipe that stale pool metadata from the partition:**
$ zpool labelclear /dev/sdb1
/dev/sdb1 is a member (ACTIVE) of pool "bootstick"

**Is it?**
$ zpool status
  pool: bootstick
 state: ONLINE
status: [your pool sucks]
action: [enable all features]
  scan: resilvered 7.19G in 0 days 00:01:38 with 0 errors
config:

        NAME        STATE     READ WRITE CKSUM
        bootstick   ONLINE       0     0     0
          sda1      ONLINE       0     0     0

errors: No known data errors

**Whatever, let's _really_ wipe it. Obey me:**
$ zpool labelclear -f /dev/sdb1
/dev/sdb1 is a member (ACTIVE) of pool "bootstick"

**WTF? Do we have an early case of raise of the machines?**
**Going termonuclear:**
$ dd if=/dev/zero of=/dev/sdb1
dd: writing to '/dev/sdb1': No space left on device
61700064+0 records in
61700063+0 records out
31590432256 bytes (32 GB, 29 GiB) copied, 188.129 s, 168 MB/s
$ zpool attach -f  -i ashift=9 bootstick /dev/sda1 /dev/sdb1
[wait]
$ zpool status
  pool: bootstick
 state: ONLINE
status: [your pool sucks]
action: [enable all features]
  scan: resilvered 7.19G in 0 days 00:01:38 with 0 errors 
config:

        NAME        STATE     READ WRITE CKSUM
        bootstick   ONLINE       0     0     0
          mirror-0  ONLINE       0     0     0
            sda1    ONLINE       0     0     0
            sdb1    ONLINE       0     0     0

$ zpool split -R /mnt/funtoo bootstick boot /dev/sdb1
cannot open '/dev/sdb1': Device or resource busy

**Yea, right... I _for some reason_ am running in reduced trust mode:**
$ zpool status
  pool: boot
 state: ONLINE
status: [your pool sucks]
action: [enable all features]
  scan: resilvered 7.19G in 0 days 00:01:38 with 0 errors
config:

        NAME        STATE     READ WRITE CKSUM
        boot        ONLINE       0     0     0
          sdb1      ONLINE       0     0     0

errors: No known data errors

  pool: bootstick
 state: ONLINE
status: [your pool sucks]
action: [enable all features]
  scan: resilvered 7.19G in 0 days 00:01:38 with 0 errors
config:

        NAME        STATE     READ WRITE CKSUM
        bootstick   ONLINE       0     0     0
          sda1      ONLINE       0     0     0

errors: No known data errors

# TL;DR: _What again was `-f` ment to do_?
Please fix these cases of disobedience.
The system should never, ever, try to discuss with, or be smarter than, _root_. 

Also ZFS should only complain about actual problems, either it should have been unable to do something on the zpool split (then this operation would have _failed_) or it should have done what it was told _without throwing a red herring_.

### Include any warning/errors/backtraces from the system logs
-none-

**Comment by GregorKopka** [comment]
Unless this is fixed: not stale.

**Comment by RJVB** [comment]
The same "resource busy" message is shown in #6643 (googling it leads to a fix committed years ago but hard to say if the spurious message was ever suppressed).

Either way, I'm bumping into that same issue right now with ZFS (onlinux) 0.8.4 . Except that it looks like there's an actual problem somewhere. This is a pool (holding my root) that sits on a disk partition, mirrored to another disk partition (a way of creating an identical backup that I've been using for years). When I try to split off the mirror I get the error message, and a few seconds later my entire pool seems to hang.

I can't remember that splitting off a pool of about 240Gb ever took longer than the few minutes I gave the command to sort itself out, but then the resilver also never took as long as it did this time (7 hours).

Should I just let it run overnight and see then? I haven't seen anything in the 0.8.5/6 changelogs that suggest a fix but maybe there is?

**Comment by GregorKopka** [comment]
Bad bot.

**Comment by behlendorf** [comment]
While I haven't manually retested this I thought we resolved this in the 2.1.x releases.
